File: lumira-app/src/data/templates/ccd-retro-portrait.ts
import type { PhotoTemplate } from '@/types/template'

// 模板：CCD 胶片复古 — 90 年代 CCD 复古质感，暖黄颗粒自带柔光
const ccdRetroPortrait: PhotoTemplate = {
  meta: {
    id: 'ccd_retro_portrait',
    name: 'CCD 胶片复古',
    author: '如画 Lumira',
    version: '1.0.0',
    category: 'portrait',
    classification: { type: 'portrait', style: 'ccd_retro', method: 'half_body' },
    tags: ['人像', 'CCD', '复古', '胶片', '暖黄'],
    tagIds: [],
    price: 0,
    cover: 'https://picsum.photos/seed/template-ccd-retro-portrait/400/600',
    description: '90 年代 CCD 复古质感，暖黄颗粒自带柔光，拍出老照片的温柔记忆',
    referenceSource: '设计规范 v1.0 模板 1；参考 vivo X200 Ultra CCD 模式 / ProCCD 教程'
  },
  composition: {
    overlayType: 'rule_of_thirds',
    subjectFrame: { x: 0.28, y: 0.15, w: 0.45, h: 0.7 },
    opacity: 0.3,
    aspectRatio: '3:4',
    description: '三分线左侧构图，半身取景，头部位于上三分线'
  },
  pose: {
    silhouette: { type: 'builtin', data: 'ccd-retro-portrait' },
    position: { x: 0.5, y: 0.5 },
    scale: 0.75,
    rotation: 0,
    description: '随性侧身回眸，一手自然下垂，一手轻触发梢，一前一后重心后移，微笑看镜头'
  },
  camera: {
    exposureCompensation: 0.3,
    iso: 200,
    shutterSpeed: '1/125',
    whiteBalance: 'cloudy',
    whiteBalanceK: 6000,
    flashMode: 'off',
    focusMode: 'auto',
    filterPreset: 'none',
    lensType: '1x'
  },
  sceneGuide: {
    lightDirection: '侧顺光 45°',
    lightDirectionAngle: 45,
    shootingDistance: '1.5-2m',
    background: '老街 / 室内暖光 / 复古墙面',
    props: [],
    bestTime: '下午 15:00-17:00',
    bestTimeFrom: '15:00',
    bestTimeTo: '17:00',
    tips: [
      '利用午后暖光营造复古氛围',
      '可轻微晃动模拟 CCD 对焦不准',
      '服装选择纯色或格纹'
    ]
  },
  postProcess: {
    cropRatio: '3:4',
    color: {
      brightness: 8,
      contrast: -5,
      saturation: 5,
      temperature: 15,
      tint: 0,
      highlights: -10,
      shadows: 10,
      clarity: -5,
      vibrance: 5
    },
    smoothStrength: 15,
    sharpen: 8,
    vignette: 15,
    grain: 20,
    lut: 'vintage'
  }
}

export default ccdRetroPortrait

File: lumira_app_flutter/lib/features/capture/data/templates/ccd_retro_portrait.dart
// lib/features/capture/data/templates/ccd_retro_portrait.dart
import '../../domain/photo_template.dart';
import '../../../templates/data/remote_template_dto.dart';

/// CCD 胶片复古模板
/// 来源：docs/superpowers/specs/2026-08-04-portrait-template-redesign-design.md 模板 1
const PhotoTemplate ccdRetroPortraitTemplate = PhotoTemplate(
  meta: TemplateMeta(
    id: 'ccd_retro_portrait',
    name: 'CCD复古半身人像',
    author: '如画 Lumira',
    version: '1.0.0',
    category: 'portrait',
    classification: TemplateClassification(type: 'portrait', majorStyle: 'emotional_film', style: 'ccd_retro', subStyle: 'ccd_retro', method: 'half_body'),
    tags: ['人像', 'CCD', '复古', '胶片', '暖黄'],
    tagIds: [],
    price: 0,
    images: [
      TemplateImage(url: 'assets/images/templates/ccd_retro_portrait.png'),
    ],
    description: '城市街头金时刻纪实人像模板，捕捉女生行走瞬间，侧逆光暖调光影，简约休闲日常穿搭，利用环境虚化前景烘托主体，适合日常出行街拍，记录松弛自然的行走动态。',
    shortDesc: '落日金辉漫洒老街，步履从容，慵懒松弛的城市氛围感✨',
    ambience: RemoteTemplateAmbienceDto(
      seasons: ['spring', 'autumn'],
      weathers: ['sunny'],
      timeTones: ['goldenHour'],
    ),
    referenceSource: '小红书 CCD 复古拍照教程；vivo X200 Ultra CCD 模式；ProCCD App 滤镜',
  ),
  composition: Composition(
    overlayType: 'rule_of_thirds',
    gridType: 'thirds',
    subjectFrame: SubjectFrame(x: 0.26, y: 0.08, w: 0.6, h: 0.91),
    opacity: 0.3,
    aspectRatio: '3:4',
    description: '全身景别，平视机位，人物落在画面垂直左三分线附近；头顶保留约1/6留白，脚底贴近画面底边；前景虚化人物占画面左右，主体人物占画幅约60%，城市街道环境占40%。',
  ),
  poses: [
    Pose(
      name: '行走动态主姿势',
      silhouette: SilhouetteResource(type: 'image', data: 'assets/images/silhouettes/ccd_retro_portrait_pose1.png'),
      position: Position(x: 0.44, y: 0.52),
      scale: 0.92,
      rotation: 0,
      description: '人物向前行走，身体朝向屏幕左约45°，面部转向屏幕左约90°，低头；屏幕右侧手插裤袋，屏幕左侧手握冰咖啡；重心落在屏幕右侧腿，屏幕左侧腿向前迈出；视线看向屏幕左下方，神情松弛淡然，帆布包搭在屏幕右肩膀。',
      cameraDirection: 'back',
    ),
  ],
  camera: CameraParams(
    exposureCompensation: -0.3,
    isoMode: 'auto',
    iso: 200,
    shutterSpeed: '1/125',
    whiteBalance: 'daylight',
    whiteBalanceK: 6200,
    flashMode: 'off',
    focusMode: 'auto',
    lensSuggestion: 'main',
    lensType: 'main',
  ),
  sceneGuide: SceneGuide(
    lightDirection: '主光从屏幕右上方45°高位侧逆光，属于黄金时刻硬质太阳光；无额外辅光；光比约3.5:1；影子投向屏幕左下方，阴影边缘锐利，地面长投影；阳光勾勒头发、肩膀轮廓，皮肤保留自然哑光纹理，墙面形成条状投影；环境背景店铺内有微弱暖黄环境光；整体主体暖金色。',
    lightDirectionAngle: 45,
    shootingDistance: '2.0-2.5m',
    background: '石质复古建筑外墙，临街咖啡馆玻璃橱窗，石板步行街路面；画面左右保留路人作为虚化前景',
    props: ['冰美式咖啡杯', '米白色帆布托特包'],
    bestTime: '16:00-17:30',
    bestTimeFrom: '15:00',
    bestTimeTo: '17:00',
    tips: [
      '抓拍行走动态，不要摆拍僵硬站姿；人物自然迈步，手部动作放松',
      '左右两侧安排人物做前景，靠近前景物体、拉大主体与背景距离，模拟样片虚化效果；手机无法实现光学大光圈虚化，依靠构图实现近似氛围',
      '利用建筑墙体条状光影，尽量让阳光打在人物身体一侧；避开正午顶光',
      '压低曝光补偿保留高光发丝轮廓，后期适当提亮暗部细节，加深暗角集中视线到人物身上',
    ],
  ),
  postProcess: PostProcess(
    cropRatio: '3:4',
    color: PostProcessColor(
      brightness: -2,
      contrast: 10,
      saturation: 8,
      temperature: 7,
      tint: 1,
      highlights: -14,
      shadows: 11,
      clarity: 6,
    ),
    smoothStrength: 28,
    sharpen: 24,
    vignette: 28,
    grain: 24,
    lut: 'warm_film',
  ),
);